python/cuda_parallel/cuda/parallel/experimental/_iterators.py

```diff
@@ -331,19 +314,21 @@ def alignment(self):
         return self.it.alignment  # TODO fix for stateful op
 
 
-def TransformIterator(op, it, op_return_ntype):
+def TransformIterator(op, it):
     # TODO(rwgk): Resolve issue #3064
 
-    op_return_ntype_ir = _ir_type_given_numba_type(op_return_ntype)
-    if op_return_ntype_ir is None:
-        raise RuntimeError(f"Unsupported: {type(op_return_ntype)=}")
     if hasattr(it, "dtype"):
         assert not hasattr(it, "ntype")
         it = pointer(it, getattr(numba.types, str(it.dtype)))
     it_ntype_ir = _ir_type_given_numba_type(it.ntype)
     if it_ntype_ir is None:
         raise RuntimeError(f"Unsupported: {type(it.ntype)=}")
 
+    op_abi_name = f"{op.__name__}_{it.ntype.name}"
+    op_ltoir = _ncc(op_abi_name, op, (it.ntype,))
+    op_return_ntype = op_ltoir[1]
```

Comment:
Oh, you need the `resty` here.

I'd probably go with another tiny helper, to avoid the zip trick:
* `_ncc()` returns `ltoir`, `resty`
* `_ncc_ltoir()` returns only `ltoir`

Reply:
Instead of adding an additional helper function, I now return a `namedtuple` from `_ncc`. If we want just the `ltoir`, we can extract that from the result as:

```python
ltoir = _ncc(...).ltoir
```